### PR TITLE
chore(*): refactor cargo crate dep

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Refactor cargo crate dependencies into
the workspace. This facilitates updating
Polkadot versions.

issue  #1751

### CODE REVIEW

File: pallets/msa/src/runtime-api/Cargo.toml [hunk 1]
```diff
@@ -14,9 +14,9 @@ codec = { package = "parity-scale-codec", version = "3.6.1", default-features =
 	"derive"
```
Comment (wilwade) [inline pallets/msa/src/runtime-api/Cargo.toml:14]: Should `codec` one line up here use workspace?

File: pallets/msa/Cargo.toml [hunk 2]
```diff
@@ -13,32 +13,31 @@ version = "0.0.0"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
-log = "0.4.17"
-codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false, features = [
+log = { workspace = true }
+parity-scale-codec = { workspace = true, features = [
```
Comment (wilwade) [inline pallets/msa/Cargo.toml:17]: I don't quite understand the `codec` vs `parity-scale-codec` swap. Don't we want the generic codec one?

File: pallets/msa/src/rpc/Cargo.toml [hunk 3]
```diff
@@ -11,22 +11,22 @@ edition = "2021"
 
 [dependencies]
 codec = { package = "parity-scale-codec", version = "3.6.1" }
```
Comment (wilwade) [inline pallets/msa/src/rpc/Cargo.toml:13]: Should this be workspace versioned?

File: common/primitives/Cargo.toml [hunk 4]
```diff
@@ -13,25 +13,25 @@ version = "0.0.0"
 targets = ['x86_64-unknown-linux-gnu']
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false, features = [
+codec = { package = "parity-scale-codec", workspace = true, features = [
```
Comment (wilwade) [inline common/primitives/Cargo.toml:16]: I'm seeing warnings that package isn't needed when I try to build.

**Review by wilwade — CHANGES_REQUESTED** [review]
Wow. This is going to be SOOO much easier for upgrades and (as I had to do recently) switching to a local build of polkadot-sdk.

Few questions, but the other 99% is 🥇

File: pallets/msa/Cargo.toml (hunk 2 above)
Comment (enddynayn) [reply pallets/msa/Cargo.toml:17]: Ahh forgot to put it back.

File: common/primitives/Cargo.toml (hunk 4 above)
Comment (enddynayn) [reply common/primitives/Cargo.toml:16]: Interestingly when I remove it fails to build. Ill take a look again.

File: Cargo.toml [hunk 5]
```diff
@@ -9,6 +9,144 @@ members = [
...
+sp-io = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
+sp-runtime = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
+sp-std = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
+sp-session = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0", default-features = false }
+sp-weights = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
+sp-offchain = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0", default-features = false }
+scale-info = { version = "2.10.0", default-features = false, features = [
+  "derive",
+] }
+sp-keystore = { default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
```
Comment (aramikm) [inline Cargo.toml:40]: nit: I think the order of features should be consistent. Like let's have all of them like `"https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0", default-features = false`  or any other order but keeping it consistent across all dependencies

File: Cargo.toml [hunk 6]
```diff
@@ -9,6 +9,144 @@ members = [
...
+[workspace.dependencies]
+log = { version = "0.4.17", default-features = false }
+env_logger = "0.10.0"
+twox-hash = { version = "1.6.3", default-features = false, features = ["digest_0_10"]}
+thiserror = "1.0.40"
+apache-avro = { version = "0.14.0", default-features = false }
+
+# substrate wasm
+codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false  }
+parity-scale-codec = { version = "3.6.1", default-features = false }
```
Comment (mattheworris) [inline Cargo.toml:21]: Here `codec === parity-scale-codec`.
Can we use one or the other of these aliases throughout our code? i.e. rename any existing `parity-scale-codec` references to `codec` and then delete the `parity-scale-codec` reference.

[The Cargo Book-Renaming dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml)

File: pallets/frequency-tx-payment/Cargo.toml [hunk 7]
```diff
@@ -13,31 +13,31 @@ version = "0.0.0"
...
 
 [dependencies]
-codec = {package = "parity-scale-codec", version = "3.6.1", default-features = false, features = [
+parity-scale-codec = { workspace = true, features = [
   "derive",
 ]}
-frame-benchmarking = {default-features = false, git = "https://github.com/paritytech/polkadot-sdk", optional = true, branch = "release-polkadot-v1.1.0" }
-frame-support = {default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
-frame-system = {default-features = false, git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.1.0" }
+frame-benchmarking = { workspace = true }
```
Comment (aramikm) [inline pallets/frequency-tx-payment/Cargo.toml:19]:
```suggestion
frame-benchmarking = { workspace = true, optional = true }
```

File: pallets/handles/Cargo.toml [hunk 8]
```diff
@@ -13,26 +13,26 @@ repository = "https://github.com/libertyDSNP/frequency/"
...
+numtoa = { workspace = true }
 # Substrate
-frame-benchmarking = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, optional = true, branch = "release-polkadot-v1.1.0" }
-frame-support = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-frame-system = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-core = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-io = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-runtime = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-std = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
+frame-benchmarking = { workspace = true }
```
Comment (aramikm) [inline pallets/handles/Cargo.toml:21]:
```suggestion
frame-benchmarking = { workspace = true, optional = true }
```

File: pallets/messages/Cargo.toml [hunk 9]
```diff
@@ -13,21 +13,21 @@ version = "0.0.0"
...
 ] }
 # Substrate
-frame-benchmarking = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, optional = true, branch = "release-polkadot-v1.1.0" }
-frame-support = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-frame-system = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-core = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-io = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-runtime = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-std = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
+frame-benchmarking = { workspace = true }
```
Comment (aramikm) [inline pallets/messages/Cargo.toml:24]:
```suggestion
frame-benchmarking = { workspace = true, optional = true }
```

File: pallets/messages/Cargo.toml [hunk 10]
```diff
@@ -38,7 +38,7 @@ multibase = { version ="0.9", default-features = false }
 common-runtime = { path = '../../runtime/common', default-features = false }
 # Testing dependencies
 rand = "0.8.5"
-pretty_assertions = "1.3.0"
+pretty_assertions = { workspace = true }
 serde = { version = "1.0", features = ["derive"] }
```
Comment (aramikm) [inline pallets/messages/Cargo.toml:42]: why some of these dependencies are kept explicitly here and not refactored into workspace?

File: pallets/stateful-storage/Cargo.toml [hunk 11]
```diff
@@ -13,29 +13,27 @@ version = "0.0.0"
...
+twox-hash = { workspace = true,  features = ["digest_0_10"] }
 # Substrate
-frame-benchmarking = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, optional = true, branch = "release-polkadot-v1.1.0" }
-frame-support = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-frame-system = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-core = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-io = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-runtime = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
-sp-std = { git = "https://github.com/paritytech/polkadot-sdk", default-features = false, branch = "release-polkadot-v1.1.0" }
+frame-benchmarking = { workspace = true }
```
Comment (aramikm) [inline pallets/stateful-storage/Cargo.toml:23]:
```suggestion
frame-benchmarking = { workspace = true, optional = true }
```

File: Cargo.toml [hunk 12]
```diff
@@ -9,6 +9,144 @@ members = [
...
 
+[workspace.dependencies]
+log = { version = "0.4.17", default-features = false }
+env_logger = "0.10.0"
+twox-hash = { version = "1.6.3", default-features = false, features = ["digest_0_10"]}
+thiserror = "1.0.40"
+apache-avro = { version = "0.14.0", default-features = false }
+
+# substrate wasm
+codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false  }
```
Comment (mattheworris) [inline Cargo.toml:20]:
```suggestion
```

**Review by aramikm — COMMENTED** [review]
Great cleanup! Added some comments and would like you to also verify the following

- If the binary size changed by this refactoring and how so?
- check the and run the changes using different features like benchmarks and etc.

File: pallets/schemas/src/runtime-api/Cargo.toml [hunk 13]
```diff
@@ -10,21 +10,17 @@ repository = "https://github.com/libertyDSNP/frequency/"
 edition = "2021"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false, features = [
```
Comment (mattheworris) [inline pallets/schemas/src/runtime-api/Cargo.toml:13]: Rename to parity-scale-codec, allows removing `codec` alias from root `Cargo.toml`

File: pallets/schemas/src/runtime-api/Cargo.toml [hunk 14]
```diff
@@ -10,21 +10,17 @@ repository = "https://github.com/libertyDSNP/frequency/"
...
+sp-api = { workspace = true }
+sp-std = { workspace = true }
+frame-support = { workspace = true }
 # Frequency related dependencies
 common-primitives = { default-features = false, path = "../../../../common/primitives" }
 
 [features]
 default = ["std"]
 std = [
-	"codec/std",
```
Comment (mattheworris) [inline pallets/schemas/src/runtime-api/Cargo.toml:27]: Rename to parity-scale-codec, allows removing `codec` alias from root `Cargo.toml`

File: pallets/schemas/src/rpc/Cargo.toml [hunk 15]
```diff
@@ -10,25 +10,25 @@ repository = "https://github.com/libertyDSNP/frequency/"
 edition = "2021"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "3.6.1" }
-jsonrpsee = { version = "0.16.2", features = ["client-core", "server", "macros"] }
+codec = { workspace = true }
```
Comment (mattheworris) [inline pallets/schemas/src/rpc/Cargo.toml:13]:
```suggestion
parity-scale-codec = { workspace = true }
```

File: pallets/messages/Cargo.toml (hunk 10 above)
Comment (enddynayn) [reply pallets/messages/Cargo.toml:42]: probably missed it

File: Cargo.toml (hunk 12 above)
[the remark this answers is not in the document]
Comment (enddynayn) [reply Cargo.toml:20]: probably left over

File: pallets/schemas/src/rpc/Cargo.toml (hunk 15 above)
Comment (enddynayn) [reply pallets/schemas/src/rpc/Cargo.toml:13]: This is probably compiled because can actually remove this. Nice catch.

**Review by mattheworris — APPROVED** [review]
Love this, nice work!

- missed one reference to `codec` in `schemas/src/rpc` 
- Executed `make ci-local`, `make benchmarks-local`

File: Cargo.toml (hunk 5 above)
Comment (enddynayn) [reply Cargo.toml:40]: sound good to me

**Review by aramikm — APPROVED** [review]
Looks good to me! My favorite kind of refactor.

**Review by wilwade — APPROVED** [review]
💯 

- All my issues are resolved
- Build is clean
- Lock file looks good.

**Pull request merged**